src/base/anim/control.cpp

```diff
@@ -150,15 +143,48 @@ void Control::finish(bool preRun)
 		}
 	}
 	else if (preRun
-	         && ((direction == Direction::normal && atEndPosition())
-	             || (direction == Direction::reverse
+	         && ((options.direction == Direction::normal
+	                 && atEndPosition())
+	             || (options.direction == Direction::reverse
 	                 && atStartPosition()))
-	         && playState != PlayState::running) {
+	         && options.playState != PlayState::running) {
 		if (!finished && onFinish) {
 			onFinish(true);
 			finished = true;
 		}
 	}
 }
 
+void Control::setValue(std::string_view path,
+    const std::string &value)
+{
+	if (path == "seek") { seek(value); }
```

Comment:
it would be better, if we could just set 'position' with duration and percent.

Reply:
I rewrite to position and set "seek" ts function